src/util/amount.rs

```diff
@@ -100,6 +108,14 @@ fn denomination_from_str(mut s: &str) -> Option<Denomination> {
         return Some(Denomination::MicroBitcoin);
     }
 
+    if s.eq_ignore_ascii_case("nBTC") {
+        return Some(Denomination::NanoBitcoin);
```

Comment:
I suggest adding inside this if:

```rust
if s.starts_with('N') {
    return None;
}
```

Reply:
How about doing this in `from_str`(https://github.com/rust-bitcoin/rust-bitcoin/pull/768/files#diff-37ff9af2290ccfd3f22d2a1e399973975fb6bd3fdae8203c223009403adb1c6eR95)?
Otherwise I feel like this private function and caller function(`from_str`) is a bit interleaving each other in responsibility.